src/journal/compress.c

```diff
@@ -231,6 +286,52 @@ int decompress_blob_lz4(const void *src, uint64_t src_size,
 #endif
 }
 
+
+int decompress_blob_zstd(const void *src, uint64_t src_size,
+                       void **dst, size_t *dst_alloc_size, size_t* dst_size, size_t dst_max) {
+#if HAVE_ZSTD
+        size_t ret;
+        char* out;
+        unsigned long long size;
+
+        assert(src);
+        assert(src_size > 0);
+        assert(dst);
+        assert(dst_alloc_size);
+        assert(dst_size);
+        assert(*dst_alloc_size == 0 || *dst);
+
+        if (src_size < ZSTD_frameHeaderSize_prefix)
+                return -EBADMSG;
+
+        if (src_size < ZSTD_frameHeaderSize(src, (size_t)src_size))
```

Comment:
You don't need this call, if the frame is invalid (or too small) `ZSTD_getFrameContentSize()` will return an error.